iceoryx_dds_gateway/include/ioxdds/gateway/channel.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+
+#include <iceoryx_posh/iceoryx_posh_types.hpp>
+#include <iceoryx_posh/popo/subscriber.hpp>
+#include <iceoryx_utils/internal/objectpool/objectpool.hpp>
+
+#include "ioxdds/dds/data_writer.hpp"
+#include "ioxdds/dds/dds_types.hpp"
+
+namespace iox
+{
+namespace gateway
```

Comment:
shouldn't this probably be first the dds namespace and then the gateway namespace?

Reply:
There are two namespaces being used.

`iox::gateway` contains iceoryx gatway implementations, of which the dds gateway is one and is included in `iox::gateway::dds`.
`iox::dds` contains utilities for interfacing with dds.

Reply:
Well, if its iox::gateway, then it would belong into iceoryx_posh. Currently we always have the namespaces like iox::module or maybe iox::module::subcomponent, so this would be different than in the remaining codebase